presto-main/src/main/java/io/prestosql/execution/QueryStateMachine.java

```diff
@@ -814,6 +814,7 @@ public boolean transitionToFailed(Throwable throwable)
 
         boolean failed = queryState.setIf(FAILED, currentState -> !currentState.isDone());
         if (failed) {
+            updateQueryInfo(Optional.empty());
```

Comment:
The point of final query info is so when a query finshes, we can immediately clean up resources and notify a client, but later when we have the root causes we fire an even with the full query info.  In the case of a failure I believe we still want to wait for final info so we can fire and event with the root causes.  After a brief look a the code, I think this disables final query info for all failure cases.  Instead, I think we should add the `updateQueryInfo` call to the caller of this method in the planner.

Reply:
Makes sense. I moved it to the caller in SqlQueryExecution.